### PR TITLE
F# API strategy decider + schedulers

### PR DESCRIPTION
**Pull request by Horusiath**
List of changes:

- New type of `IDecider` using F# quotation expressions and reflected this change in `Strategy` methods inside F# API.
- Completely new version of supervision strategy example to include actor lifecycle presentation - since F# actors don't expose lifcycle hook methods explicitly, new example presents how equivalents of the `PreStart`/`PostStop` methods could be achieved.
- (**breaking change!**) Also due to changes made by @HCanber all scheduling functions has been removed from F# API and replaced with extension methods to `IScheduler` and `IAdvancedScheduler` interfaces, which could work smoothly with F# functions and optional parameters.
- Last change (**also breaking one**) is making ask operator `<?` generic. Now it leverages F# type inference - this has been reflected in examples, which now have reduced number of explicit casts and type declarations.

cc #680 and #722 (fix proven in test case)

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Horusiath is this good to go?

**Comment by Horusiath** [comment]
Yes. It is.

File: src/core/Akka.FSharp/Schedulers.fs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿[<AutoOpen>]
+module Schedulers
+
+open Akka.Actor
+open System
+
+type Akka.Actor.IAdvancedScheduler with
```
Comment (HCanber) [inline src/core/Akka.FSharp/Schedulers.fs]: Better to change this to `type Akka.Actor.IActionScheduler`

File: src/core/Akka.FSharp/Schedulers.fs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /// <param name="after">Delay before calling the function.</param>
+    /// <param name="fn">Function called by the scheduler.</param>
+    /// <param name="cancelable">Optional cancelation token</param>
+    member this.ScheduleOnce(after: TimeSpan, fn: unit->unit, ?cancelable: ICancelable) : unit =
+        let action = Action fn
+        match cancelable with
+        | Some c -> this.ScheduleOnce(after, action, c)
+        | None -> this.ScheduleOnce(after, action)
+        
+type Akka.Actor.IScheduler with
```
Comment (HCanber) [inline src/core/Akka.FSharp/Schedulers.fs]: Change to `type Akka.Actor.ITellScheduler with`

**Pull request merged**